index.js

```diff
@@ -1,33 +1,57 @@
 const fetch = require('node-fetch'); // Automatically excluded in browser bundles
 
-async function api(endpoint, token) {
-	token = token ? `&access_token=${token}` : '';
-	const response = await fetch(`https://api.github.com/repos/${endpoint}${token}`);
+// Matches '/<user>/<repo>/tree/<ref>/<dir>'
+const urlParserRegex = /^[/]([^/]+)[/]([^/]+)[/]tree[/]([^/]+)[/](.*)/;
+
+function parseGithubURL(url) {
+	const res = {};
+	const parsedUrl = new URL(url);
+	[, res.user, res.repository, res.ref, res.directory] = urlParserRegex.exec(parsedUrl.pathname) || [];
+	if (typeof res.directory !== 'string') {
+		// eslint-disable-next-line unicorn/prefer-type-error
+		throw new Error('Unable to parse GitHub URL');
```

Comment:
```suggestion
		throw new TypeError('Unable to parse GitHub URL');
```

Reply:
I wasn't sure whether it was correct to call it a TypeError. Ok for me.